crates/release-automation/src/crate_.rs

```diff
@@ -176,7 +184,139 @@ pub(crate) fn cmd(args: &crate::cli::Args, cmd_args: &CrateArgs) -> CommandResul
 
             Ok(())
         }
+        CrateCommands::DetectMissingReleaseheadings(subcmd_args) => {
+            cmd_detect_missing_releaseheadings(&ws, subcmd_args)
+        }
+    }
+}
+
+fn cmd_detect_missing_releaseheadings<'a>(
+    ws: &'a ReleaseWorkspace<'a>,
+    _subcmd_args: &CrateDetectMissingReleaseheadings,
+) -> Fallible<()> {
+    let missing_headings = detect_missing_releaseheadings(ws)?;
+
+    if !missing_headings.is_empty() {
+        bail!("missing crate release headings: {:#?}", missing_headings);
     }
+
+    Ok(())
+}
+
+/// if there are any crate release headings present in the workspace changelog but missing from the crate changelogs an error is returned.
+///
+/// uses the workspace changelog as a source of truth for existing crate releases.
+/// this reasonable because the workspace changelog it's only changed on releases it's not prone to manual mistakes.
+pub(crate) fn detect_missing_releaseheadings<'a>(
+    ws: &'a ReleaseWorkspace<'a>,
+) -> Fallible<LinkedHashMap<String, LinkedHashSet<String>>> {
+    use itertools::Itertools;
+
+    let crtheadings_toplevel = {
```

Comment:
I stumble upon a name like this and prefer spelling it out `crate_headings_toplevel`
or does `crt` mean something else?